generators/entity/index.js

```diff
@@ -1204,6 +772,404 @@ class EntityGenerator extends BaseBlueprintGenerator {
+        if (entity.applicationType === 'gateway' && entity.useMicroserviceJson) {
+            if (!entity.microserviceName) {
+                throw new Error('Microservice name for the entity is not found. Entity cannot be generated!');
+            }
+            entity.microserviceAppName = this.getMicroserviceAppName(entity.microserviceName);
+            entity.skipServer = true;
+        }
+
+        entity.entityNameCapitalized = _.upperFirst(entity.name);
+        entity.entityClass = entity.entityNameCapitalized;
+        entity.entityClassPlural = pluralize(entity.entityClass);
+
+        // Used for i18n
+        entity.entityClassHumanized = entityConfig.entityClassHumanized || _.startCase(entity.entityNameCapitalized);
+        entity.entityClassPluralHumanized = entityConfig.entityClassPluralHumanized || _.startCase(entity.entityClassPlural);
+        // Implement i18n variant ex: 'male', 'female' when applied
+        entity.entityI18nVariant = entityConfig.entityI18nVariant || 'default';
+
+        entity.entityInstance = _.lowerFirst(entityName);
+        entity.entityInstancePlural = pluralize(entity.entityInstance);
+
+        entity.entityFileName = _.kebabCase(entity.entityNameCapitalized + _.upperFirst(entity.entityAngularJSSuffix));
+        entity.entityFolderName = this.getEntityFolderName(entity.clientRootFolder, entity.entityFileName);
+        entity.entityModelFileName = entity.entityFolderName;
+        entity.entityParentPathAddition = this.getEntityParentPathAddition(entity.clientRootFolder);
+        entity.entityPluralFileName = entityNamePluralizedAndSpinalCased + entity.entityAngularJSSuffix;
+        entity.entityServiceFileName = entity.entityFileName;
+
+        entity.entityAngularName = entity.entityClass + this.upperFirstCamelCase(entity.entityAngularJSSuffix);
+        entity.entityReactName = entity.entityClass + this.upperFirstCamelCase(entity.entityAngularJSSuffix);
+
+        entity.entityApiUrl = entityNamePluralizedAndSpinalCased;
+        entity.entityStateName = _.kebabCase(entity.entityAngularName);
+        entity.entityUrl = entity.entityStateName;
+
+        entity.entityTranslationKey = entity.clientRootFolder
+            ? _.camelCase(`${entity.clientRootFolder}-${entity.entityInstance}`)
+            : entity.entityInstance;
+        entity.entityTranslationKeyMenu = _.camelCase(
+            entity.clientRootFolder ? `${entity.clientRootFolder}-${entity.entityStateName}` : entity.entityStateName
+        );
+
+        entity.reactiveRepositories = entity.reactive && ['mongodb', 'cassandra', 'couchbase', 'neo4j'].includes(entity.databaseType);
+
+        entity.differentTypes.push(entity.entityClass);
+        entity.i18nToLoad.push(entity.entityInstance);
+        entity.i18nKeyPrefix = `${entity.angularAppName}.${entity.entityTranslationKey}`;
+
+        entity.hasUserField = !!entity.relationships.find(relationship => this.isBuiltInUserEntity(relationship.otherEntityName));
```

Comment:
See https://github.com/jhipster/generator-jhipster/pull/12176/files#diff-c026af5fb08143a94ab3665821943885L855 : this property is not used in template, not need to save it.
```suggestion
        let hasUserField = !!entity.relationships.find(relationship => this.isBuiltInUserEntity(relationship.otherEntityName));
```